servers/rest-services/src/main/java/org/projectnessie/services/rest/RestTreeResource.java

```diff
@@ -65,8 +71,41 @@ private TreeService resource() {
   @JsonView(Views.V1.class)
   @Override
   public ReferencesResponse getAllReferences(ReferencesParams params) {
-    Preconditions.checkArgument(params.pageToken() == null, "Paging not supported");
-    return resource().getAllReferences(params.fetchOption(), params.filter());
+    Integer maxRecords = params.maxRecords();
+    return resource()
+        .getAllReferences(
+            params.fetchOption(),
+            params.filter(),
+            null,
+            new PagedResponseHandler<
+                ImmutableReferencesResponse.Builder, ReferencesResponse, Reference>() {
+              final int max = maxRecords != null ? Math.max(maxRecords, 0) : 0;
+
+              @Override
+              public ImmutableReferencesResponse.Builder newBuilder() {
+                return ReferencesResponse.builder();
```

Comment:
Do we need to pass the builder through the service code? Why not create it above and refer to it via a local var? In the end we have 1:1 between the builder and the service invocation, right?

Reply:
removed